src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs

```diff
@@ -2005,7 +2005,8 @@ private ISwitchOperation CreateBoundPatternSwitchStatementOperation(BoundSwitchS
             ITypeSymbol type = null;
             Optional<object> constantValue = default(Optional<object>);
             bool isImplicit = boundPatternSwitchStatement.WasCompilerGenerated;
-            return new LazySwitchStatement(locals, value, cases, exitLabel, _semanticModel, syntax, type, constantValue, isImplicit);
+            bool wasPatternSwitch = boundPatternSwitchStatement.Expression.Type?.IsValidV6SwitchGoverningType() == false;
+            return new LazySwitchStatement(locals, value, cases, exitLabel, wasPatternSwitch, _semanticModel, syntax, type, constantValue, isImplicit);
```

Comment:
>wasPatternSwitch [](start = 76, length = 16)

Please add regular IOperation tree tests for scenarios when ```wasPatternSwitch``` is true.

Reply:
wasPatternSwitch has been eliminated.

---
In reply to: [233161044](https://github.com/dotnet/roslyn/pull/31129#discussion_r233161044) [](ancestors = 233161044)